### PR TITLE
Add --dev switch

### PR DESCRIPTION
**Pull request by josephwright**

### CODE REVIEW

**Comment by muzimuzhi** [comment]
Need to also add `latex-dev` to where all supported `checkformat` values are listed:

https://github.com/latex3/l3build/blob/3457c13ccf94588237dd7f501d82f31f1af6b990/l3build.dtx#L434-L436

**Comment by josephwright** [comment]
@muzimuzhi Done

**Review by davidcarlisle — APPROVED** [review]

**Review by muzimuzhi — APPROVED** [review]

**Comment by FrankMittelbach** [comment]
I think this functionality warrants a ltnews entry (in a different repo I know) explaining how a maintainer of a package using l3build for it can easily test his or her package with the upcoming LaTeX release this way.

File: l3build.lua [hunk 1]
```diff
@@ -131,6 +131,10 @@ if forcedocepoch then
     typesetopts = typesetopts .. " -utc"
   end
 end
+-- Allow for LaTeX "dev" release
+if options["dev"] and checkformat == "latex" then
+  checkformat = "latex-dev"
```
Comment (FrankMittelbach) [inline l3build.lua:136]: not sure if it helps or if it makes things worse, but shouldn't there perhaps a terminal output stating --dev not applied if checkformat is not latex?

**Review by FrankMittelbach — APPROVED** [review]

File: l3build.lua (hunk 1 above)
Comment (josephwright) [reply l3build.lua]: I've added one - it only shows at the top level due to the way configs work, but hopefully that's enough?

**Pull request merged**